language-plutus-core/src/Language/PlutusCore/Type.hs

```diff
@@ -82,6 +88,16 @@ instance Corecursive (Type tyname a) where
     embed (TyLamF l tn k ty)    = TyLam l tn k ty
     embed (TyAppF l ty ty')     = TyApp l ty ty'
 
+-- | Get all the direct child 'Type's of the given 'Type'.
+typeSubtypes :: Traversal' (Type tyname a) (Type tyname a)
+typeSubtypes f = fmap embed . traverse f . project
+
+-- | Get all the direct child 'tyname a's of the given 'Type' from 'TyVar's.
+typeTyVars :: Traversal' (Type tyname a) (tyname a)
```

Comment:
Should it be a `Prism`?

Reply:
I guess it could be, but I'm not sure that helps much?

Reply:
Juts a bit more useful: you can check whether a type is variable using a prism and maybe something else. Not really important.